### ISSUE TITLE
.NET Core November 2017 Update

### ISSUE THREAD

**Issue by richlander** [dotnet/core#1082]
# .NET Core September 2017 Update

The .NET Core November 2017 Update is available. This release includes .NET Core 1.0.8 .NET Core 1.1.5 and 2.0.1 and .NET Core SDK 1.1.5 and 2.0.3.

You can read about the update in [announcement](https://blogs.msdn.microsoft.com/dotnet/2017/11/14/net-core-november-2017-update/) or the [.NET Core release notes](https://github.com/dotnet/core/blob/master/release-notes/README.md).

Please report any issues you find with the November 2017 Update, either responding to this issue, creating a new issue or creating a new issue in one of the following repos:

* [dotnet/cli](https://github.com/dotnet/cli/issues) - for CLI tools and questions
* [dotnet/corefx](https://github.com/dotnet/corefx/issues) - for API issues and questions
* [dotnet/coreclr](https://github.com/dotnet/coreclr/issues) - for runtime issues
* [dotnet/core-setup](https://github.com/dotnet/core-setup/issues) - for installer issues
* [nuget/home](https://github.com/nuget/home/issues) - for NuGet questions and issues

**Comment by martincostello** [comment]
The download link for the 32-bit SDK binaries (https://download.microsoft.com/download/5/C/1/5C190037-632B-443D-842D-39085F02E1E8/dotnet-sdk-2.0.3-win-x86.zip) does not appear to work.

![image](https://user-images.githubusercontent.com/1439341/32826868-fc860a74-c9e1-11e7-9281-cffc5d6e1c9a.png)

**Comment by leecow** [comment]
Sorry about that @martincostello , fixed.

**Comment by acjh** [comment]
Issue title and heading says *September*, but description and announcement says *November*.

**Comment by richlander** [comment]
Thanks on that @acjh. You can see that I'm living in the past.

**Comment by acjh** [comment]
No worries, but the heading still says *September*.

**Comment by DavBrennan** [comment]
Although dotnet-sdk-2.0.3 rpm is in the repository it does not appear to be set to upgrade previous installs. So **yum update** does not install the new version. I noted that dotnet-host 2.0.3 has been updated automatically. Is this correct? Can 2.0.3 sdk be installed without removing 2.0.0

**Comment by Petermarcu** [comment]
Yes, 2.0.3 can be installed without removing 2.0.0. Everything targeting 2.0 will start to run on 2.0.3 unless they have explicitly said they don't want to run on anything higher than 2.0.0.

It would be nice if we could still have yum update give the new version even though its installed side by side with the old one.

**Comment by DavBrennan** [comment]
Not really going to be a problem we package our apps as rpms so can require 2.0.3 to get it installed. I assume from the above that 2.0.0 can be removed without hurting 2.0.3

**Comment by Petermarcu** [comment]
Yes, 2.0.0 can be removed unless you have applications that have specifically locked themselves to 2.0.0 which should be rare to never. :)

**Comment by leecow** [comment]
@DavBrennan - enabling yum/dnf/apt-get update behavior is on the list to work through. Because .NET Core is entirely side-by-side, it presents a few interesting challenges which will need to be figured out.

**Comment by DavBrennan** [comment]
@leecow - you could create an empty parent rpm say dotnet-update that uses weak Recommends instead of Requires. That would pull new versions of the sdk but would also allow old versions to be removed. See https://fedoraproject.org/wiki/Packaging:WeakDependencies

**Comment by leecow** [comment]
Excellent, thanks for the pointer @DavBrennan ! fyi @rakeshsinghranchi

**Comment by Petermarcu** [comment]
Whatever we do, we should apply to both runtimes and SDK's if there is a good way to do it.

**Comment by CRCinAU** [comment]
With the packaging for CentOS, did the location of the dotnet binary change between dotnet-host-2.0.0-1.x86_64 and dotnet-host-2.0.3-1.x86_64?

I noticed that 'dotnet' is not in any of the default paths after the upgrade, but the binary is present as /usr/share/dotnet/dotnet

This broke every one of our apps that use 'dotnet' as included in the system path.

I'm not sure where the spec files are for the creation of these packages, so I can't take a look at how it was done...

**Comment by Petermarcu** [comment]
All the packaging for the runtime is in the dotnet/core-setup repo

**Comment by rakeshsinghranchi** [comment]
https://github.com/dotnet/core-setup/tree/master/src/pkg/packaging/rpm/scripts . There is an issue with uninstall script that's causing major upgrade remove the symbolic link . Binary RPMs are built using FPM tool.

**Comment by CRCinAU** [comment]
Hrrrm - in the .spec world, you'd do something like this:
```
%preun
if [ $1 = 0 ]; then
   unlink /usr/bin/dotnet
fi
```

This roughly translates to: If we're uninstalling (hence %preun), and we don't have any newer version installed, run the script.

Running a complete script file in what I assume is the %post or %preun sections of the RPM installation - I'm not sure there is an easy way to achieve the same - unless you can pass the same $1 to the script and evaluate it in said script...

**Comment by chenowen** [comment]
I got an asp.net application which is using 2.0.0, and im try to upgrade to 2.0.3. 
The 2.0.3 SDK has intall, but i can't upgrade Microsoft.NetCore.App in Nuget. It says i need to upgrade my SDK, which i already did.

What am i missing?

**Comment by Petermarcu** [comment]
@chenowen , what are you trying to update? A setting in your project file? Are you publishing a self-contained application or a shared framework application? 

If its a shared framework application, it will run on the highest patched version of 2.0.X. So if you installed the 2.0.3 SDK that installs the 2.0.3 runtime and your shared application will run against it.

**Comment by rakeshsinghranchi** [comment]
@chenowen , Thanks. Yes, the proposed fix needs an update in https://github.com/dotnet/core-setup/blob/master/src/pkg/packaging/rpm/scripts/after_remove_host.sh . Opened https://github.com/dotnet/core-setup/issues/3455 to track this issue.

**Comment by chenowen** [comment]
@Petermarcu I use vs 15.4 to create the project, but the project file missing one configuration <RuntimeFrameworkVersion/>. I have to open the project file and add it manually to fix the issue.

**Comment by Petermarcu** [comment]
I guess the next question is why you need to upgrade the NuGet package? Are you building a self-contained application?

**Comment by chenowen** [comment]
@Petermarcu Because of i want to upgrade to EF core 2.0.3

**Comment by Petermarcu** [comment]
I see, you shouldn't need to update netcoreapp in order to upgrade EF Core. Do you have a reference to:

`<PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.0" />`

In your project? That is the reference that you can update to 2.0.3 in order to get EF Core updated?

**Comment by chenowen** [comment]
@Petermarcu That doesn't help. 

![image](https://user-images.githubusercontent.com/1118166/33695964-fb739fc4-db3a-11e7-9490-ad881618fca4.png)

![image](https://user-images.githubusercontent.com/1118166/33695979-0d52c85a-db3b-11e7-831c-305c98e3fd70.png)

I have to add `<RuntimeFrameworkVersion/>` to upgrade to 2.0.3.

**Comment by Ming-Sizung** [comment]
@chenowen I am able to confirm (using the gist (here)[https://gist.github.com/npnelson/5bf49a145338120cacbac7ee8bd399bf]) even the visual studio reports that the project is running against dotnet core version 2.0.0, the actual runtime is still going to be the latest version, aka 2.0.3.

@Petermarcu I can see it is very confusing for developers who struggle to confirm if everything is up-to-date. Would love to see this display version get fixed.

**Comment by Petermarcu** [comment]
When you are building a shared framework application, your application can run on 2.0.X for any version of X. Its not until you deploy to production that you know exactly what you are running on.

@Ming-Sizung where in VS are you seeing the display version?

**Comment by Ming-Sizung** [comment]
@Petermarcu I understand it should be the real behavior. But if you go to "Manage NuGet Packages" for a project (targeting `netcoreapp20`), it will still report its running against `Microsoft.NetCore.App` version `2.0.0` instead of `2.0.3`.

**Comment by chenowen** [comment]
@Petermarcu The issue is VS think the project is using dotnet core version 2.0.0, so it can't upgrade the ef core to 2.0.3. As ef core 2.0.3 require dotnet core 2.0.3.

**Comment by EriGH** [comment]
Hello, after updating to DotNetCore.1.0.8_1.1.5-WindowsHosting.exe, installation removed the AspNetCoreModule from IIS modules on Windows 2012 R2 Server and I'm unable to run IIS hosted appliaction. Previous version is OK. Does anyone have the same problem? Thank you

**Comment by Petermarcu** [comment]
Closing in favor of https://github.com/dotnet/core/issues/1172
